types/validator_set.go

```diff
@@ -660,12 +664,21 @@ func (vals *ValidatorSet) VerifyCommit(chainID string, blockID BlockID,
 
 		// Validate signature.
 		voteSignBytes := commit.VoteSignBytes(VotePrefix(chainID, vals.Hash()), idx)
-		if !val.PubKey.VerifyBytes(voteSignBytes, commitSig.Signature) {
-			return fmt.Errorf("wrong signature (#%d): %X", idx, commitSig.Signature)
-		}
+
 		// Good!
-		if blockID.Equals(commitSig.BlockID(commit.BlockID)) {
+		if commitSig.ForBlock() {
 			talliedVotingPower += val.VotingPower
+			blsKey, ok := val.PubKey.(bls12_381.PubKeyBls)
+			if !ok {
+				panic(fmt.Sprintf("incorrect key type for combined signatures"))
```

Comment:
this can only happen if everyone commits to a block that has a non bls sig, which everyone should have rejected earlier, right?

Reply:
Just thinking about ways that everyone could end up panicing

Reply:
Yes, I think for the combined sig stuff we need to ensure when new validators are created that the keys are all bls keys.